cypress/integration/unit_tests/polygon_draw_test.js

```diff
@@ -176,6 +177,38 @@ describe('Unit test for ShapeData', () => {
     });
   });
 
+  it('Shows calculating before update finishes', () => {
+    const div = document.createElement('div');
+    document.getElementsByTagName('body')[0].appendChild(div);
+    const map = new google.maps.Map(div, {center: {lat: 0, lng: 0}, zoom: 1});
+    const polygon = new google.maps.Polygon({
+      map: map,
+      paths: [{lat: 0, lng: 0}, {lat: 1, lng: 1}, {lat: 0, lng: 1}],
+    });
+    const event = new Event('overlaycomplete');
+    event.overlay = polygon;
+    firebaseCollection.add = () => Promise.resolve({id: 'id'});
+    const updatePromise = new SettablePromise();
+    cy.wrap(setUpPolygonDrawing(map, Promise.resolve()))
+        .then((drawingManager) => {
+          google.maps.event.trigger(drawingManager, 'overlaycomplete', event);
+          updatePromise.setPromise(event.resultPromise);
+        });
+    let calculatingDiv;
+    // Popup initialization happens async, apparently, and takes a bit of time.
+    cy.wait(50).then(() => {
```

Comment:
Sorry, I'm not quite understanding - what is ensuring that these expects run before the calculating finishes?

Reply:
Added a comment.